packages/react/src/primitives/shared/styleUtils.ts

```diff
@@ -105,33 +128,49 @@ export const convertStylePropsToStyleObj: ConvertStylePropsToStyleObj = ({
   breakpoints,
   tokens,
 }) => {
-  const nonStyleProps = {};
+  const nonStyleProps: Record<string, unknown> = {};
   Object.keys(props)
-    .filter((propKey) => props[propKey] != null)
-    .forEach((propKey: ThemeStylePropKey) => {
-      if (!(propKey in ComponentPropsToStylePropsMap)) {
-        nonStyleProps[propKey] = props[propKey] as ViewProps;
-      } else if (!isEmptyString(props[propKey])) {
-        const values = props[propKey] as ViewProps;
-        const reactStyleProp = ComponentPropsToStylePropsMap[propKey];
+    .filter((propKey) => props[propKey] !== null)
+    .forEach((propKey) => {
+      if (isComponentStyleProp(propKey)) {
+        const values = props[propKey];
+        if (!values || isEmptyString(values)) return;
 
-        style = {
-          ...style,
-          [reactStyleProp]: getValueAtCurrentBreakpoint({
-            values,
-            breakpoint,
-            breakpoints,
+        const reactStyleProp = ComponentPropsToStylePropsMap[propKey];
+        // short circuit the style prop here if it is a string or design token
+        // so we don't have to call getValueAtCurrentBreakpoint every time
+        let value: number | string | null = '';
+        if (isDesignToken(values)) {
```

Comment:
Has nothing to do with this PR, but we should probably rename `isDesignToken` to is `isWebDesignToken`